lib/streamlit/elements/widgets/button_group.py

```diff
@@ -300,77 +331,153 @@ def feedback(
             transformed_options,
             default=None,
             key=key,
-            click_mode=ButtonGroupProto.SINGLE_SELECT,
+            selection_mode="single",
             disabled=disabled,
             deserializer=serde.deserialize,
             serializer=serde.serialize,
             on_change=on_change,
             args=args,
             kwargs=kwargs,
             selection_visualization=selection_visualization,
+            style="borderless",
         )
         return sentiment.value
 
+    @gather_metrics("pills")
+    def pills(
+        self,
+        label: str,
+        options: OptionSequence[V],
+        *,
+        selection_mode: Literal["single", "multi"] = "single",
+        icons: list[str | None] | None = None,
+        default: Sequence[V] | V | None = None,
+        format_func: Callable[[V], str] | None = None,
+        key: str | int | None = None,
+        help: str | None = None,
+        on_change: WidgetCallback | None = None,
+        args: WidgetArgs | None = None,
+        kwargs: WidgetKwargs | None = None,
+        disabled: bool = False,
+        label_visibility: LabelVisibility = "visible",
+    ):
+        maybe_raise_label_warnings(label, label_visibility)
+
+        def _transformed_format_func(
+            option: V, icon: str | None = None
+        ) -> ButtonGroupProto.Option:
+            if format_func is None:
+                return ButtonGroupProto.Option(content=str(option), content_icon=icon)
+
+            transformed = format_func(option)
+            return ButtonGroupProto.Option(
+                content=transformed,
+                content_icon=icon,
+            )
+
+        indexable_options = convert_to_sequence_and_check_comparable(options)
+        default_values = get_default_indices(indexable_options, default)
+
+        serde = MultiSelectSerde(indexable_options, default_values)
+        res = self._button_group(
+            indexable_options,
+            key=key,
+            icons=icons,
+            default=default_values,
+            selection_mode=selection_mode,
+            disabled=disabled,
+            format_func=_transformed_format_func,
+            serializer=serde.serialize,
+            deserializer=serde.deserialize,
+            on_change=on_change,
+            args=args,
+            kwargs=kwargs,
+            style="pills",
+            label=label,
+            label_visibility=label_visibility,
+            help=help,
+        )
+
+        if selection_mode == "multi" and len(res.value) > 0:
```

Comment:
nit: do we need to check `res.value` before `len(res.value)`? like L406

Reply:
great catch! Though I think its the other way around. The type of `res` is `list` and not `list | None`, so it should never be `None` and I guess we should rely on our typing system or update it. What do you think?

Reply:
ahh that's true! make senses to remove this for single selection